stdlib/tkinter/simpledialog.pyi

```diff
@@ -1,10 +1,13 @@
-from tkinter import Event, Misc, Toplevel
-from typing import Any
+from tkinter import Event, Frame, Misc, Toplevel
 
 class Dialog(Toplevel):
     def __init__(self, parent: Misc | None, title: str | None = ...) -> None: ...
-    def body(self, master) -> None: ...
+    def body(self, master: Frame) -> Misc | None: ...
     def buttonbox(self) -> None: ...
+    def ok(self, event: Event[Misc] | None = ...) -> None: ...
+    def cancel(self, event: Event[Misc] | None = ...) -> None: ...
```

Comment:
The events come from `self.bind(...)`, so their widget is always the type of `Self` (so a `Dialog` or some subclass of `Dialog`):

```suggestion
    def ok(self: Self, event: Event[Self] | None = ...) -> None: ...
    def cancel(self: Self, event: Event[Self] | None = ...) -> None: ...
```

You will need to import `Self` from `_typeshed`.

Reply:
I did consider constraining this further, but since the `buttonbox` method (adding the buttons) is intended to be overridden, in that case it seems reasonable that the user might want to bind these to other widgets, so it would no longer apply. The default implementations don't use the widget at all, and in either case you can still access `self` directly. I should have put a comment to explain that though.